### ISSUE TITLE
Add stepper component to Mesop

### ISSUE THREAD

**Issue by emc5ud** [google/mesop#114]
The stepper component in Angular Material is very useful for designing flows where the user needs to input a variety of data into a form step by step.  

https://material.angular.io/components/stepper/overview

**Comment by richard-to** [comment]
The Stepper component does seem useful. I think the one thing we'll need a solution for is how the content inside the different steps will work when written as code and also how that will be implemented internally in core Mesop.

I think there is a similar thing with the tab component (https://github.com/google/mesop/issues/93).

I assume it is possible because we can render components within components. The difference is that now we have components within components that would need to work together.

**Comment by emc5ud** [comment]
This implementation in the "dash mantine components" library has separate object types for the "stepper" and the "stepperstep"

https://www.dash-mantine-components.com/components/stepper

**Comment by richard-to** [comment]
Yeah I agree. I think that would be the direction to go. We had some discussion about this in the context of the tab component as well. So I think we're closer to how the API could look for these kind of multiple component components.
